shell/platform/darwin/macos/framework/Source/AccessibilityBridgeMacDelegate.mm

```diff
@@ -0,0 +1,377 @@
+            .target = native_node,
+            .user_info = nil,
+        });
+      }
+      break;
+    case ui::AXEventGenerator::Event::DOCUMENT_SELECTION_CHANGED: {
+      // This event always fires at root
+      events.push_back({
+          .name = NSAccessibilitySelectedTextChangedNotification,
+          .target = native_node,
+          .user_info = nil,
+      });
+      // WebKit fires a notification both on the focused object and the page
+      // root.
+      const ui::AXTreeData& tree_data = bridge->GetAXTreeData();
+      int32_t focus = tree_data.focus_id;
+      if (focus == ui::AXNode::kInvalidAXID || focus != tree_data.sel_anchor_object_id) {
+        break;  // Just fire a notification on the root.
+      }
+      auto focus_node = bridge->GetFlutterPlatformNodeDelegateFromID(focus).lock();
+      if (!focus_node) {
+        break;  // Just fire a notification on the root.
+      }
+      events.push_back({
+          .name = NSAccessibilitySelectedTextChangedNotification,
+          .target = focus_node->GetNativeViewAccessible(),
+          .user_info = nil,
+      });
+      break;
+    }
+    case ui::AXEventGenerator::Event::CHECKED_STATE_CHANGED:
+      events.push_back({
+          .name = NSAccessibilityValueChangedNotification,
+          .target = native_node,
+          .user_info = nil,
+      });
+      break;
+    case ui::AXEventGenerator::Event::VALUE_CHANGED:
+      events.push_back({
+          .name = NSAccessibilityValueChangedNotification,
+          .target = native_node,
+          .user_info = nil,
+      });
+      if (@available(macOS 10.11, *)) {
+        if (ax_node.data().HasState(ax::mojom::State::kEditable)) {
+          events.push_back({
+              .name = NSAccessibilityValueChangedNotification,
+              .target = bridge->GetFlutterPlatformNodeDelegateFromID(kRootNode)
+                            .lock()
+                            ->GetNativeViewAccessible(),
+              .user_info = nil,
+          });
+        }
+      }
+      break;
+    case ui::AXEventGenerator::Event::LIVE_REGION_CREATED:
+      events.push_back({
+          .name = AccessibilityLiveRegionCreatedNotification,
+          .target = native_node,
+          .user_info = nil,
+      });
+      break;
+    case ui::AXEventGenerator::Event::ALERT: {
+      events.push_back({
+          .name = AccessibilityLiveRegionCreatedNotification,
+          .target = native_node,
+          .user_info = nil,
+      });
+      // Voiceover requires a live region changed notification to actually
+      // announce the live region.
+      auto live_region_events =
+          MacOSEventFromAXEvent(ui::AXEventGenerator::Event::LIVE_REGION_CHANGED, ax_node);
+      events.insert(events.end(), live_region_events.begin(), live_region_events.end());
+      break;
+    }
+    case ui::AXEventGenerator::Event::LIVE_REGION_CHANGED: {
+      if (@available(macOS 10.14, *)) {
+      } else {
```

Comment:
Invert the check logic here and place this in the if block, then kill off the else block.

Reply:
I got compiler error when i did
```objc
 if (!@available(macOS 10.14, *)) {
}
```
it seems like it has to be by  if (@available(..., ...))  inorder to use this annotation?